pip/vcs/git.py

```diff
@@ -240,4 +272,19 @@ def update_submodules(self, location):
             cwd=location,
         )
 
+    @classmethod
+    def controls_location(cls, location):
+        if super(Git, cls).controls_location(location):
+            return True
+        try:
+            r = Git().run_command(['rev-parse'],
```

Comment:
`cls()` seems cleaner than explicitly using `Git()` (even if `Git` is unlikely to be subclassed)